src/snowflake/snowpark/dataframe.py

```diff
@@ -611,6 +636,29 @@ def to_local_iterator(
     def __copy__(self) -> "DataFrame":
         return DataFrame(self._session, copy.copy(self._plan))
 
+    if installed_pandas:
```

Comment:
What will happen to the IDE/notebook when pandas is not installed?

Reply:
It's the same as before. Again @overload is for docstring and type checking tools so this PR doesn't have any functional change. When pandas is not installed, only this definition will be skipped, and users are not able to get autocomplete for pandas methods.